### ISSUE TITLE
Disabling auth breaks the admin view

### ISSUE THREAD

**Issue by spacemanspiff2007** [openhab/openhab-core#2094]
Disabling auth as described by @ghys breaks the administration view.

Steps to reproduce:
Add this to ```userdata/etc/org.apache.karaf.features.xml``` and clear ```userdata/cache```.

```yml
    <blacklistedFeatures>
        <feature>openhab-*-auth</feature>
    </blacklistedFeatures>
```
Restart openhab and try the administration button:

![grafik](https://user-images.githubusercontent.com/10754716/104115271-cfbecb80-530d-11eb-98f9-1ce040e10a30.png)

**Comment by kaikreuzer** [comment]
> Disabling auth as described by @ghys breaks the administration view.

I understood @ghys' suggestion as a workaround to try out. Doing such settings is not any official feature, this would only be the case if we have configuration settings in place.

**Comment by spacemanspiff2007** [comment]
> I understood @ghys' suggestion as a workaround to try out.

To be fair there was no mentioning this being a workaround and the arguing against a setting made it seem like the only way to do this.
Anyway I pinged you in the other issue and it would be really nice if you could give me a hint or some help.
Thanks in advance.

**Comment by ghys** [comment]
I only answered your question about preventing the auth features from being installed, I never said it was a supported scenario. The UI doesn't support not having auth in place.

**Comment by spacemanspiff2007** [comment]
My intention is not to point fingers or blame someone, I am just looking for a solution and therefore I am thankful for any help.
I think it's obvious that the current implementation is somehow broken and it would be nice we could join up and fix things.
Imho there are mainly two things left to tackle:
- make auth opt out
- fix the performance issues with basic auth

For the latter I think I might have something in the pr.

**Comment by kaikreuzer** [comment]
> I think it's obvious that the current implementation is somehow broken

Not sure if this is obvious.
The only concrete issue I have seen is https://github.com/openhab/openhab-core/issues/1899 and I agree that those performance issue should be fixed. I haven't seen any clear clue yet on what exactly is causing them, because [as mentioned](https://github.com/openhab/openhab-core/pull/2030#discussion_r550923148), I don't think that hashing should be the issue in general.
If auth would only have insignificant overhead, is there anything else that I am missing in the discussion?

**Comment by Pedals2Paddles** [comment]
I'm a little surprised this isn't a higher priority.  OpenHAB 3 is **completely unusable from outside the LAN where authentication is required**.  It seems like having no access to this system over the internet would be a pretty big deal.  If any of the other big HA systems were not accessibly from the internet in some way, it would be the end for them.

It can't be just me and spaceman that are the only ones that use OpenHAB remotely?

**Comment by kaikreuzer** [comment]
> completely unusable from outside the LAN where authentication is required

Sounds like a very subjective statement. Everyone using myopenhab or openhab-cloud or using VPN definitely have no issue at all.
And for people using NGINX as a reverse proxy, it should be fairly similar to add a basic auth header for the requests to the proxy host. What am I missing?

**Comment by spacemanspiff2007** [comment]
> And for people using NGINX as a reverse proxy, it should be fairly similar to add a basic auth header for the requests to the proxy host. What am I missing?

It is related performance issue. The performance measurements were done on my main machine and basic auth already had a very significant and unacceptable performance impact. On the rasperry pi or lower end machines basic auth is basically useless because it just takes too long and then congests. One user of HABApp reported that instead of 100ms its now 30s for him.
Imagine flipping a switch in openhab and it takes 30 secs for the light to turn on.
Of doing a gui refresh through the reverse proxy where loading the gui takes longer than a minute.

**Comment by Pedals2Paddles** [comment]
Is there actually anyone who thinks that the MyopenHAB Cloud has acceptable performance? It is unacceptably slow and clunky for routine use. It's pretty well documented every time someone complains, that it is a free resource, it is what it is, and if you don't like it, use NGINX with basic auth. And that's okay, it is a free resource which I am absolutely not complaining about.

But the documented and recommended alternative method for remote access through NGINX using basic auth is 100% unusable right now because of this API auth situation. If you disable the API auth, everything starts working again But you cannot log in as an administrator.

Suggesting that all of our users establish themselves a personal VPN, routing all of their mobile device internet traffic through their house, is not an acceptable or reasonable solution. We all know that's not a thing that we can expect everyone to do, nor should we.

So it is not what I would consider an unreasonable and subjective statement to say that openhab is unusable remotely right now.

**Comment by spacemanspiff2007** [comment]
I have completed and verified a fix for the issue: #2101
